src/sentry/tasks/app_store_connect.py

```diff
@@ -41,42 +42,70 @@ def inner_dsym_download(
     client = appconnect.AppConnectClient.from_config(config)
     itunes_client = client.itunes_client()
 
-    builds = client.list_builds()
-    for build in builds:
-        try:
-            build_state = AppConnectBuild.objects.get(
-                project=project,
-                app_id=build.app_id,
-                platform=build.platform,
-                bundle_short_version=build.version,
-                bundle_version=build.build_number,
-            )
-        except AppConnectBuild.DoesNotExist:
-            build_state = AppConnectBuild(
-                project=project,
-                app_id=build.app_id,
-                bundle_id=config.bundleId,
-                platform=build.platform,
-                bundle_short_version=build.version,
-                bundle_version=build.build_number,
-                fetched=False,
-            )
-
+    # persist all fetched builds into the database as "pending"
+    builds = []
+    for build in client.list_builds():
+        build_state = get_or_create_persisted_build(project, config, build)
         if not build_state.fetched:
-            with tempfile.NamedTemporaryFile() as dsyms_zip:
-                try:
-                    itunes_client.download_dsyms(build, pathlib.Path(dsyms_zip.name))
-                except appconnect.NoDsymsError:
-                    logger.debug("No dSYMs for build %s", build)
-                    continue
+            builds.append(build_state)
+
+    for build_state in builds:
+        with tempfile.NamedTemporaryFile() as dsyms_zip:
+            try:
+                itunes_client.download_dsyms(build, pathlib.Path(dsyms_zip.name))
+
                 create_difs_from_dsyms_zip(dsyms_zip.name, project)
-            build_state.fetched = True
-            build_state.save()
-            logger.debug("Uploaded dSYMs for build %s", build)
+                logger.debug("Uploaded dSYMs for build %s", build)
+            except appconnect.NoDsymsError:
+                logger.debug("No dSYMs for build %s", build)
+            except FailedFetchingDsymError:
+                # TODO: We should probably persist this error in some way so
+                # we can inform the user about expired itunes credentials.
```

Comment:
We already have specced out that we want to show pending builds.  We also already show expired credentials.  So I think this is already covered no?